python/setup.py

```diff
@@ -194,7 +194,15 @@ def is_pure(self):
     curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
     for i, path in enumerate(LIB_LIST):
         LIB_LIST[i] = os.path.relpath(path, curr_path)
-    setup_kwargs = {"include_package_data": True, "data_files": [("tvm", LIB_LIST)]}
+    LIB_DATA_FILES = []
+    if os.path.isfile(path):
+        LIB_DATA_FILES.append(os.path.relpath(path, curr_path))
+    if os.path.isdir(path):
+        for root, dirs, files in os.walk(path):
+            for filename in files:
+                file_path = os.path.join(root, filename)
+                LIB_DATA_FILES.append(os.path.relpath(file_path, curr_path))
+    setup_kwargs = {"include_package_data": True, "data_files": [("tvm", LIB_DATA_FILES)]}
```

Comment:
The logic here seems duplicated with `f"recursive-include tvm/{libname} *\n"` above. Is there a way to unify so that we can only add files in one standard way that works for local installation as well as packaging?

Reply:
I have made some change for this, please help me do a double check. I found the `data_files` filed is valid for both `bdist_wheel` and other actions, so I think it was no need to create a conditional branch.
I have tested `python setup.py bdist_wheel` and `python setup.py install`, and they created the same library files.

Reply:
I think the issue with that approach is that internal subpaths are lost and the files are kept flat in the package. Can you send the output of `tree` on the directory you extract from the `.egg / .wheel` files?

Reply:
Yes, you are right, the files are kept flat in the package. I try to find out the best practice of packing non-package file in setup.py and this PR gives my idea: https://github.com/pypa/sampleproject/issues/30

While include_package_data=True, the setup program will go to run command in MANIFEST.in. This step is explained in [setuptools/docs/references/keywords.rst](https://github.com/pypa/setuptools/blob/main/docs/references/keywords.rst)

> include_package_data: If set to True, this tells setuptools to automatically include any data files it finds inside your package directories that are specified by your MANIFEST.in file. For more information, see the section on [:ref:`Including Data Files`](https://github.com/pypa/setuptools/blob/0bd847c4aea839b3a75f4e3f09a36526453bd267/docs/references/keywords.rst#id23).

In the first PR: https://github.com/apache/tvm/pull/380 for data_files field, I found the include_package_data was set to False in wheel packing. So the detail of include_package_data may be ignored by the commiter.

Here is the result of unzipping .egg / .whl file:

.whl 
```
dist
├── tvm
│   ├── __init__.py
│   ├── _ffi
│   ├── arith
│   ├── auto_scheduler
│   ├── autotvm
│   ├── configs
│   ├── contrib
│   ├── driver
│   ├── error.py
│   ├── exec
│   ├── generic.py
│   ├── ir
│   ├── libtvm.so
│   ├── libtvm_runtime.so
│   ├── meta_schedule
│   ├── micro
│   ├── parser
│   ├── relay
│   ├── rpc
│   ├── runtime
│   ├── script
│   ├── support.py
│   ├── target
│   ├── te
│   ├── testing
│   ├── tir
│   ├── topi
│   └── utils
├── tvm-0.9.dev1753+g6236f2cb6-cp38-cp38-linux_x86_64.whl
└── tvm-0.9.dev1753+g6236f2cb6.dist-info
    ├── METADATA
    ├── RECORD
    ├── WHEEL
    ├── entry_points.txt
    └── top_level.txt
```

.egg

```
dist
├── EGG-INFO
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   ├── dependency_links.txt
│   ├── entry_points.txt
│   ├── native_libs.txt
│   ├── not-zip-safe
│   ├── requires.txt
│   └── top_level.txt
├── tvm
│   ├── __init__.py
│   ├── _ffi
│   ├── arith
│   ├── auto_scheduler
│   ├── autotvm
│   ├── configs
│   ├── contrib
│   ├── driver
│   ├── error.py
│   ├── exec
│   ├── generic.py
│   ├── ir
│   ├── libtvm.so
│   ├── libtvm_runtime.so
│   ├── meta_schedule
│   ├── micro
│   ├── parser
│   ├── relay
│   ├── rpc
│   ├── runtime
│   ├── script
│   ├── support.py
│   ├── target
│   ├── te
│   ├── testing
│   ├── tir
│   ├── topi
│   └── utils
└── tvm-0.9.dev1754+g2a71c380e-py3.8-linux-x86_64.egg
```

Reply:
Understood. Can you build TVM with `USE_MICRO ON`, so that more files are expected to be put in the package, namely `standalone_crt`, which has a deeper file structure?

Reply:
the unzipped .egg file:
```
.
├── EGG-INFO
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   ├── dependency_links.txt
│   ├── entry_points.txt
│   ├── native_libs.txt
│   ├── not-zip-safe
│   ├── requires.txt
│   └── top_level.txt
├── tvm
│   ├── __init__.py
│   ├── _ffi
│   ├── arith
│   ├── auto_scheduler
│   ├── autotvm
│   ├── configs
│   ├── contrib
│   ├── driver
│   ├── error.py
│   ├── exec
│   ├── generic.py
│   ├── ir
│   ├── libtvm.so
│   ├── libtvm_runtime.so
│   ├── meta_schedule
│   ├── micro
│   ├── microtvm_template_projects
│   ├── parser
│   ├── relay
│   ├── rpc
│   ├── runtime
│   ├── script
│   ├── standalone_crt
│   ├── support.py
│   ├── target
│   ├── te
│   ├── testing
│   ├── tir
│   ├── topi
│   └── utils
└── tvm-0.9.dev1754+g2a71c380e-py3.8-linux-x86_64.egg
```

the `standalone_crt` folder:
```
(base) ➜ tree standalone_crt -L 2
standalone_crt
├── Makefile
├── host
│   └── Makefile
├── include
│   ├── checksum.h
│   ├── dlpack
│   ├── dmlc
│   └── tvm
├── src
│   ├── runtime
│   └── support
└── template
    ├── crt_config-template.h
    └── host
```

Reply:
That looks great! Thanks @Codle!